Tests/OAuth/ResourceOwner/AppleResourceOwnerTest.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/*
+ * This file is part of the HWIOAuthBundle package.
+ *
+ * (c) Hardware Info <opensource@hardware.info>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace HWI\Bundle\OAuthBundle\Tests\OAuth\ResourceOwner;
+
+use HWI\Bundle\OAuthBundle\OAuth\ResourceOwner\AppleResourceOwner;
+use HWI\Bundle\OAuthBundle\Tests\Fixtures\CustomUserResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+class AppleResourceOwnerTest extends GenericOAuth2ResourceOwnerTest
+{
+    protected $resourceOwnerClass = AppleResourceOwner::class;
+    protected $userResponse = <<<json
+{
+    "sub": "1",
+    "email": "localhost@gmail.com"
+}
+json;
+
+    protected $paths = [
+    'identifier' => 'sub',
+    'firstname' => 'firstName',
+    'lastname' => 'lastName',
+    'email' => 'email',
+  ];
+
+    protected $expectedUrls = [
+    'authorization_url' => 'http://user.auth/?test=2&response_type=code&client_id=clientid&scope=name+email&redirect_uri=http%3A%2F%2Fredirect.to%2F&response_mode=form_post',
+    'authorization_url_csrf' => 'http://user.auth/?test=2&response_type=code&client_id=clientid&scope=name+email&state=random&redirect_uri=http%3A%2F%2Fredirect.to%2F&response_mode=form_post',
+  ];
+
+    public function testHandleRequest()
+    {
+        $request = new Request(['test' => 'test']);
+
+        $this->assertFalse($this->resourceOwner->handles($request));
+
+        $request = new Request(['code' => 'test']);
+
+        $this->assertFalse($this->resourceOwner->handles($request));
+
+        $request = new Request([], ['code' => 'test']);
+
+        $this->assertTrue($this->resourceOwner->handles($request));
+
+        $request = new Request([], ['code' => 'test', 'test' => 'test']);
+
+        $this->assertTrue($this->resourceOwner->handles($request));
+    }
+
+    public function testGetAccessTokenFailedResponse()
+    {
+        $this->expectException(\Symfony\Component\Security\Core\Exception\AuthenticationException::class);
+
+        $this->mockHttpClient('{"error": {"message": "invalid"}}', 'application/json; charset=utf-8');
+        $request = new Request(['code' => 'code']);
+
+        $this->resourceOwner->getAccessToken($request, 'http://redirect.to/');
+    }
+
+    public function testDisplayPopup()
+    {
+        $resourceOwner = $this->createResourceOwner($this->resourceOwnerName, ['display' => 'popup']);
+
+        $this->assertEquals(
+      $this->options['authorization_url'].'&response_type=code&client_id=clientid&scope=name+email&state=random&redirect_uri=http%3A%2F%2Fredirect.to%2F&response_mode=form_post',
+      $resourceOwner->getAuthorizationUrl('http://redirect.to/')
```

Comment:
Not sure what happened but the indentation of a lot of files is not correct.

Reply:
I fixed indentation now in AppleResourceOwner.php and AppleResourceOwnerTest.php. In other files indentation should be ok.